testing/unit_tests/test_loader.py

```diff
@@ -216,5 +216,51 @@ def test_2d_patches(download_data_testing_test_files,
     assert len(ds) == 16
 
 
+@pytest.mark.parametrize('loader_parameters', [{
+    "path_data": [os.path.join(__data_testing_dir__, "microscopy_png")],
+    "bids_config": f"{path_repo_root}/ivadomed/config/config_bids.json",
+    "target_suffix": ["_seg-myelin-manual"],
+    "extensions": [".png"],
+    "roi_params": {"suffix": None, "slice_filter_roi": None},
+    "contrast_params": {"contrast_lst": [], "balance": {}},
+    "slice_axis": "axial",
+    "slice_filter_params": {"filter_empty_mask": False, "filter_empty_input": True},
+    "multichannel": False
+    }])
+@pytest.mark.parametrize('model_parameters', [{
+    "name": "Unet",
+    "dropout_rate": 0.3,
+    "bn_momentum": 0.1,
+    "depth": 2
+    }])
+@pytest.mark.parametrize('transform_parameters', [{
+    "NumpyToTensor": {},
+    }])
+def test_get_target_filename_list(loader_parameters, model_parameters, transform_parameters):
+    """
+    Test that all target_suffix are considered for target filename when list
+    """
+    loader_parameters.update({"model_params": model_parameters})
+    bids_df = imed_loader_utils.BidsDataframe(loader_parameters, __tmp_dir__, derivatives=True)
+    data_lst = ['sub-rat3_ses-01_sample-data9_SEM.png']
+    test_ds = imed_loader.load_dataset(bids_df,
+                                  **{**loader_parameters, **{'data_list': data_lst,
+                                                             'transforms_params': transform_parameters,
+                                                             'dataset_type': 'training'}})
+
+    test_subject = 'sub-unf02_T2star.nii.gz'
```

Comment:
The subject should be `sub-rat3_ses-01_sample-data9_SEM.png` from the data_lst above (so it is the same as the files in the dataframe)